
Careful QC of genotyping data is an important step toward a successful GWAS. We need to exclude samples and/or variants that does not meet our quality expectations. 

## 4.1. Identifcation of 1st degree relatives 

Genotyping data can be used to assess [sample-distances](./ext_docs/sample_distance.md) and thus identify sample duplications. 
We will use 0.177 (the geometric mean of 0.25 and 0.125) as cutoff for the KING knickship  coeffiecients to identify 1st degree relatives.

```
mkdir -p gwas
plink2 --bfile AxiomGT1v2.comp_merge.deDup.sexConfirm.grls_ids --chr-set 38 no-xy --allow-extra-chr \
       --king-cutoff 0.177 \
       --out gwas/AxiomGT1v2.1st_degree_relatives
```

Let us remove those relatives to avoid inflation of false associations in our GWAS

```
plink2 --bfile AxiomGT1v2.comp_merge.deDup.sexConfirm.grls_ids --chr-set 38 no-xy --allow-extra-chr \
       --remove gwas/AxiomGT1v2.1st_degree_relatives.king.cutoff.out.id \
       --make-bed --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives
```

## 4.2. Explore

We use Plink commands to assess heterozygosity, missing genotyping, allele frequency, and hardy-weinberg equilibrium (HWE). The output are the files ending in ".het", ".imiss (Per-individual)/.lmiss (per-variant)", ".frq", and ".hwe" 

```
plink --bfile gwas/AxiomGT1v2.noRelatives --chr-set 38 no-xy --allow-extra-chr \
      --het --missing --freq --hardy 'midp'  \
      --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.explore

# We currentlky have:
# 913984 variants loaded from .bim file.
# 2519 samples (1279 males, 1240 females) loaded from .fam.
```

>  *Note:* The 'midp' modifier of the command `--hardy` applies the mid-p adjustment described [here](https://www.ncbi.nlm.nih.gov/pubmed/23934608). The mid-p adjustment tends to bring the null rejection rate in line with the nominal p-value, and also reduces the filter's tendency to favor retention of variants with missing data

## 4.2.1. Extreme cases of heterozygosity:

```
awk -v size=0.02 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($6/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i){if(i==0) print -1*size,size,a[i]/1;else if(i<0) print (i-1)*size,i*size,a[i]/1;else print i*size,(i+1)*size,a[i]/1 }}'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.het) > gwas/AxiomGT1v2.noRelatives.explore.het.histo 

cat gwas/AxiomGT1v2.noRelatives.explore.het | awk '{if($6>0.3)print}' >  gwas/AxiomGT1v2.noRelatives.explore.het.high ## high homozygosity

cat gwas/AxiomGT1v2.noRelatives.explore.het | awk '{if(NR==1)print}{if($6<-0.3)print}' >  gwas/AxiomGT1v2.noRelatives.explore.het.low ## low homozygosity
```
We can see at least 2 extreme cases. High heterozygosity may indicate sample contamination, however high homozygosity might just indicate higher inbreeding


## 4.2.2. Genotype Missingness in samples and variants:

```
awk -v size=0.02 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($6/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.imiss) > gwas/AxiomGT1v2.noRelatives.explore.imiss.histo 

awk -v size=0.02 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($5/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.lmiss) > gwas/AxiomGT1v2.noRelatives.explore.lmiss.histo 
```
Missingness per individual shows 13 samples with about ~57% genotyping rates. These samples were genotyped on one array but not the other. While missingness per variant shows 36 extreme variants with about 50% genotyping rates

## 4.2.3. Variants with very low MAF:

```
awk -v size=0.01 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($5/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.frq) > gwas/AxiomGT1v2.noRelatives.explore.frq.histo 

awk -v size=0.001 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($5/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.frq) > gwas/AxiomGT1v2.noRelatives.explore.frq.histo2 
```
There are 331344 markers that have < 1% MAF, 278461 markers have < 0.1% MAF & 314175 markers have < 0.5% MAF


## 4.2.4. Markers with significant diviation from HWE

```
awk 'BEGIN{OFS="\t";}{ if($9<1e-50)a["1e-50 or less"]++;
                       else if($9<1e-40)a["1e-40:1e-50"]++; else if($9<1e-30)a["1e-30:1e-40"]++; \
                       else if($9<1e-20)a["1e-20:1e-30"]++; else if($9<1e-10)a["1e-10:1e-20"]++; \
                       else if($9<1e-9)a["1e-9:1e-10"]++; else if($9<1e-8)a["1e-8:1e-9"]++; \
                       else if($9<1e-7)a["1e-7:1e-8"]++; else if($9<1e-6)a["1e-6:1e-7"]++; \
                       else if($9<1e-5)a["1e-5:1e-6"]++; else if($9<1e-4)a["1e-4:1e-5"]++; \
                       else if($9<0.001)a["1e-3:1e-4"]++; else if($9<0.01)a["1e-2:1e-3"]++; } \
                 END { for(i in a) print i,a[i] }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.explore.hwe) | sort -g > gwas/AxiomGT1v2.noRelatives.explore.hwe.histo

## Let's have a look on those variants with extreme diviation from HWE
cat gwas/AxiomGT1v2.noRelatives.explore.hwe | awk '{if(NR==1)print}{if($9<1e-50 )print}' >  gwas/AxiomGT1v2.noRelatives.explore.hwe.low 
```

Serious genotyping errors often yield extreme p-values, therefore a cut-off like 1e-50 is usually advised to exclude these variants. However, I believe this is ok in populations where reasonable random mating is expected.  
Dog breeds are subjected to high selective breeding and artificial selection and thus I prefer excluding cases with extreme heterozygosity instead of apply any HWE-based filtering 

## 4.3. Filter
```
echo "Oldies 094-040059" > gwas/het_excess.lst
plink --bfile gwas/AxiomGT1v2.noRelatives --chr-set 38 no-xy --allow-extra-chr \
      --geno 0.05 --mind 0.05 --maf 0.01 \
      --remove gwas/het_excess.lst\
      --make-bed --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered
```
Now, we have 582519 variants and 2505 samples that passed filters and QC.


## 4.4. linkage disequilibrium (LD) analysis 

## 4.4.1 LD estimation

*    --r2 reports squared inter-variant allele count correlations
*    'in-phase' adds a column with in-phase allele pairs
*    'dprime' adds the absolute value of Lewontin's D-prime statistic

```
plink --bfile gwas/AxiomGT1v2.noRelatives.filtered --chr-set 38 no-xy --allow-extra-chr \
      --r2 'in-phase' 'dprime' \
      --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered.stats

awk -v size=0.01 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($8/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.filtered.stats.ld) > gwas/AxiomGT1v2.noRelatives.filtered.stats.ld.r2_histo ## 128035 sequential markers have complete linkage
awk -v size=0.01 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($9/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.filtered.stats.ld) > gwas/AxiomGT1v2.noRelatives.filtered.stats.ld.dprime_histo
```

## 4.4.2 LD pruning

We will prune pairwise SNPs with R2 >â€‰ 0.2 (using windows of 100 SNPs and step size of 10 SNPs)

```
plink --bfile gwas/AxiomGT1v2.noRelatives.filtered --chr-set 38 no-xy --allow-extra-chr \
      --indep-pairwise 100 10 0.2 \
      --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered.LD_lst ## 567586 of 582519 variants removed.

plink --bfile gwas/AxiomGT1v2.noRelatives.filtered --chr-set 38 no-xy --allow-extra-chr \
      --extract gwas/AxiomGT1v2.noRelatives.filtered.LD_lst.prune.in \
      --make-bed --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered.LD_prune ## 14933 variants remaining
```

## 4.4.3 LD re-evaluation
```
plink --bfile gwas/AxiomGT1v2.noRelatives.filtered.LD_prune --chr-set 38 no-xy --allow-extra-chr \
      --freq --r2 'in-phase' 'dprime' \
      --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats

awk -v size=0.01 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($8/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.ld) > gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.ld.r2_histo

awk -v size=0.01 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($9/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.ld) > gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.ld.dprime_histo

awk -v size=0.001 'BEGIN{OFS="\t";bmin=bmax=0}{ b=int($5/size); a[b]++; bmax=b>bmax?b:bmax; bmin=b<bmin?b:bmin } END { for(i=bmin;i<=bmax;++i) print i*size,(i+1)*size,a[i]/1 }'  <(tail -n+2 gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.frq) > gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.frq.histo2 ## 278461 markers have < 0.1% MAF & 314175 markers have < 0.5% MAF

paste gwas/AxiomGT1v2.noRelatives.filtered.stats.ld.r2_histo gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.stats.ld.r2_histo | less
```

## 4.5. PCA

The `--pca` function in Plink2, similar to GCTA, extracts top principal components from the variance-standardized relationship matrix (i.e. genomic relationship matrix)

```
plink2 --bfile gwas/AxiomGT1v2.noRelatives.filtered.LD_prune --chr-set 38 no-xy --allow-extra-chr \
       --autosome --pca \
       --output-chr 'chrM' --out gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.pca
```

I dentify the principle components that explain most of the variance: 
```
Rscript -e 'val <- read.table("gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.pca.eigenval");'\
'val$varPerc <- val$V1/sum(val$V1);'\
'jpeg(file = "Var_PCs.jpg");'\
'plot( x = seq(1:length(val$varPerc)), y = val$varPerc, type = "o",xlab = "Principle Component", ylab = "Variance explained in %");'\
'dev.off();'
```

![](images/Var_PCs.jpg)<!-- -->



Plot the main principle components:
```
Rscript -e 'require(ggplot2);require(gridExtra);'\
'eigenvec <- read.table("gwas/AxiomGT1v2.noRelatives.filtered.LD_prune.pca.eigenvec", header = TRUE, comment.char="");'\
'eigenvec$X.FID <-  as.factor(eigenvec$X.FID);'\
'plot1 <- ggplot(eigenvec, aes(x = PC1, y = PC2, col = X.FID)) + geom_point() + labs(title = "PCA Plot", x = "PC1", y = "PC2");'\
'plot2 <- ggplot(eigenvec, aes(x = PC3, y = PC4, col = X.FID)) + geom_point() + labs(title = "PCA Plot", x = "PC3", y = "PC4");'\
'plot3 <- ggplot(eigenvec, aes(x = PC1, y = PC3, col = X.FID)) + geom_point() + labs(title = "PCA Plot", x = "PC1", y = "PC3");'\
'plot4 <- ggplot(eigenvec, aes(x = PC2, y = PC3, col = X.FID)) + geom_point() + labs(title = "PCA Plot", x = "PC2", y = "PC3");'\
'combined_plot <- grid.arrange(plot1, plot2, plot3, plot4, nrow = 2);'\
'ggsave("pca_plot.png", combined_plot, width = 12, height = 8, dpi = 400);'
```

![](images/pca_plot.png)<!-- -->


An additional tutorial of possible admixture analysis is [here](./ext_docs/admixture_analysis.md) 
