
Our phenotype of interest will define the GWAS design. Some phenotypes are quantitative like body mass index or fasting blood glucose while other are qualitative, binary phenotypes (e.g., Case/Control studies). Each type requires special preparation of the input data, proper selection of the statistical tests, and mindful interpretation of the final results. In this tutorial, we have an example for a Case/Control study.  

To start, we have to select our cases and control based on the inclusion and exclusion criteria in our project. Accordingly, we should subset our genotypic and phenotypic data. Then, we need to assess the possible correlation of our phenotype with the genetic principal components in the studied population.  Also, we have to identify the possible covariates that might affect our phenotype of interest but we will do it during the assessment of heritability of out phenotype in the studied population. 

## 5.1. Identify affected cases until last year

Most of the input files are designed to have 2 rows for each new year of study. The 1st row records the new phenotypes diagnosed only in that study year while the 2nd row records all phenotypes diagnosed from baseline through that study year. The column `to_date` is encoded 0 for the former and 1 for the latter. You can read more about the design of each table in the [Data Commons](https://datacommons.morrisanimalfoundation.org/) of Morris Animal Foundation.

```
# Define and create the GWAS output folder
gwas="gwas_output" && mkdir -p "$gwas"

# Identify the 2nd row of the last year of study for each dog
tissue="skin"
cat phenotypes/conditions_${tissue}.csv | awk 'BEGIN{FS=","}\
    NR==1{print $0;next}{if ($5==1) {\
    if (!($1 in max_year) || $3 > max_year[$1]) {
        max_year[$1] = $3;
        line[$1] = $0;
    }}}END{for (id in line) {print line[id];}}' | tr ',' '\t' > "$gwas"/conditions_${tissue}_lastYear.tab
```

## 5.2. Explore the numbers of cases for each phenotype
```
# Get the number of columns in the header
num_columns=$(head -n 1 "$gwas/conditions_${tissue}_lastYear.tab" | awk -F"\t" '{print NF}')

# Loop over each column from the third till the last, and print the sum
for ((i=7; i<=$num_columns; i++)); do
    awk -v col="$i" 'BEGIN{FS=OFS="\t"}NR==1{dis=$col;next}{sum+=$col} END {print dis,sum}' "$gwas"/conditions_${tissue}_lastYear.tab
done > "$gwas"/conditions_${tissue}.no_of_cases
```

## 5.3. Sample selection
## 5.3.1 Select "cases" of a target phenotype
```
target="atopy"
Tcol=$(head -n 1 "$gwas/conditions_${tissue}_lastYear.tab" | awk -F"\t" -v pheno="$target" '{for (i=7; i<=NF; i++){if($i==pheno)print i}}')
awk -v Tcol="$Tcol" 'BEGIN{FS=OFS="\t"}NR==1{print;next}{if($Tcol=="1")print}' "$gwas"/conditions_${tissue}_lastYear.tab > "$gwas"/conditions_${target}_lastYear.tab

tail -n+2 "$gwas"/conditions_${target}_lastYear.tab | cut -f1 > "$gwas"/${target}_cases.ids
```

## 5.3.2 Check for co-existing conditions
```
for ((i=7; i<=$num_columns; i++)); do
    awk -v col="$i" 'BEGIN{FS=OFS="\t"}NR==1{dis=$col;next}{sum+=$col} END {if(sum)print dis,sum}' "$gwas"/conditions_${target}_lastYear.tab
done > "$gwas"/conditions_${target}_lastYear_coexist.tab
```
For now, we will not exclude any samples from cases


## 5.3.3. Identify individuals to be excluded from controls
```
grep 'allergic_reaction\|seasonal_allergy\|angioedema\|facial_edema\|.*_dermatitis\|r_o_atopy\|vaccine_reaction' "$gwas"/conditions_${tissue}.no_of_cases > "$gwas"/${target}.to_be_excluded.lst
while read offTarget;do
  offTcol=$(head -n 1 "$gwas"/conditions_${tissue}_lastYear.tab | awk -F"\t" -v pheno="$offTarget" '{for (i=7; i<=NF; i++){if($i==pheno)print i}}')
  awk -F"\t" -v Tcol="$Tcol" -v offTcol="$offTcol" 'NR>1{if($Tcol!="1" && $offTcol=="1")print $1}' "$gwas"/conditions_${tissue}_lastYear.tab
done < <(cut -f1 "$gwas"/${target}.to_be_excluded.lst) | sort | uniq > "$gwas"/${target}.to_be_excluded.ids
```

## 5.4. Remove the excluded samples from the genotyping dataset (include the oldies if not testing a phenotype with high mortality in young age)
```
cat "$gwas"/${target}.to_be_excluded.ids | grep -Fwf - AxiomGT1v2.noRelatives.filtered.LD_prune.fam > "$gwas"/${target}.to_be_excluded.samples
grep ^Oldies AxiomGT1v2.noRelatives.filtered.LD_prune.fam >> "$gwas"/${target}.to_be_excluded.samples

plink --bfile AxiomGT1v2.noRelatives.filtered.LD_prune --chr-set 38 no-xy --allow-extra-chr \
      --remove "$gwas"/${target}.to_be_excluded.samples --maf 0.01 \
      --make-bed --output-chr 'chrM' --out "$gwas"/AxiomGT1v2.noRelatives.filtered.LD_prune.${target}
```
There are 14795 variants and 1932 samples that passed filters and QC.


## 5.5. generate the phenotype file
```
awk 'FNR==NR{a[$1]=1;next}{if(!a[$2])print $1,$2,"1";else print $1,$2,"2";}' "$gwas"/${target}_cases.ids "$gwas"/AxiomGT1v2.noRelatives.filtered.LD_prune.${target}.fam > "$gwas"/${target}.pheno
cat "$gwas"/${target}.pheno | cut -d" " -f3 | sort | uniq -c ## we have 326 cases and 1606 controls
```

## 5.6. Assess distribution of cases in PCA
```
plink2 --bfile "$gwas"/AxiomGT1v2.noRelatives.filtered.LD_prune.${target} --chr-set 38 no-xy --allow-extra-chr \
       --autosome --pca \
       --output-chr 'chrM' --out "$gwas"/AxiomGT1v2.noRelatives.filtered.LD_prune.${target}.pca

Rscript -e 'args=(commandArgs(TRUE));require(ggplot2);require(gridExtra);'\
'eigenvec <- read.table(paste(args[1],"/AxiomGT1v2.noRelatives.filtered.LD_prune.",args[2],".pca.eigenvec",sep=""), header = TRUE, comment.char="");'\
'pheno <- read.table(paste(args[1],"/",args[2],".pheno",sep=""), header = FALSE, comment.char="");'\
'names(pheno) <- c("FID", "IID", "pheno");'\
'pheno$pheno[pheno$pheno == "1"] <- "Unaffected";pheno$pheno[pheno$pheno == "2"] <- "Affected";'\
'ph_eigenvec <-  merge(pheno, eigenvec, by = "IID");'\
'ph_eigenvec$pheno <-  as.factor(ph_eigenvec$pheno);'\
'plot1 <- ggplot(ph_eigenvec, aes(x = PC1, y = PC2, col = pheno)) + geom_point() + labs(title = "PCA Plot", x = "PC1", y = "PC2");'\
'plot2 <- ggplot(ph_eigenvec, aes(x = PC3, y = PC4, col = pheno)) + geom_point() + labs(title = "PCA Plot", x = "PC3", y = "PC4");'\
'plot3 <- ggplot(ph_eigenvec, aes(x = PC1, y = PC3, col = pheno)) + geom_point() + labs(title = "PCA Plot", x = "PC1", y = "PC3");'\
'plot4 <- ggplot(ph_eigenvec, aes(x = PC2, y = PC3, col = pheno)) + geom_point() + labs(title = "PCA Plot", x = "PC2", y = "PC3");'\
'combined_plot <- grid.arrange(plot1, plot2, plot3, plot4, nrow = 2);'\
'ggsave(paste(args[1],"/pca_plot_",args[2],".png",sep=""), combined_plot, width = 12, height = 8, dpi = 400);' "$gwas" "${target}"
```

![](images/pca_plot_atopy.jpg)<!-- -->

